api/service/syncing/syncing.go

```diff
@@ -290,18 +311,22 @@ func (sc *SyncConfig) GetBlockHashesConsensusAndCleanUp() bool {
 }
 
 // GetConsensusHashes gets all hashes needed to download.
-func (ss *StateSync) GetConsensusHashes(startHash []byte) bool {
+func (ss *StateSync) GetConsensusHashes(startHash []byte, size uint32) bool {
 	count := 0
 	for {
 		var wg sync.WaitGroup
 		ss.syncConfig.ForEachPeer(func(peerConfig *SyncPeerConfig) (brk bool) {
 			wg.Add(1)
 			go func() {
 				defer wg.Done()
-				response := peerConfig.client.GetBlockHashes(startHash)
+				response := peerConfig.client.GetBlockHashes(startHash, size)
 				if response == nil {
 					return
 				}
+				if len(response.Payload) > int(size+1) {
```

Comment:
Per robustness principle, be liberal in what you accept.  Just accept the first `size` and discard the rest.

Reply:
added the logic